### ISSUE TITLE
Rank With 'method=first' Broken for Objects

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#19482]
Came across this working on #15779

```python
In []: df = pd.DataFrame({'key': ['a'] * 5, 'val': ['bar', 'bar', 'foo', 'bar', 'baz']})
In []: df.groupby('key').rank(method='first')

Out []: 
Empty DataFrame
Columns: []
Index: []

```

#### Expected Output

```python

Out[]: 
   val
0  1.0
1  2.0
2  5.0
3  3.0
4  4.0

```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: d3f7d2a666aa824e2df98083aa5c1fd9bb63252e
python: 3.6.3.final.0
python-bits: 64
OS: Darwin
OS-release: 17.4.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.0.dev0+169.gd3f7d2a66.dirty
pytest: 3.2.1
pip: 9.0.1
setuptools: 36.5.0.post20170921
Cython: 0.26.1
numpy: 1.13.3
scipy: 1.0.0
pyarrow: 0.8.0
xarray: 0.10.0
IPython: 6.2.1
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: 0.4.0
matplotlib: 2.1.1
openpyxl: 2.5.0b1
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.1.13
pymysql: 0.7.11.None
psycopg2: None
jinja2: 2.10
s3fs: 0.1.2
fastparquet: 0.1.3
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by WillAyd** [comment]
Alternately this could throw a ValueError, which seems to be the intention of the code in algos linked below. That said, I don't see why it would be bad to support this and couldn't find any comment history touching on it. Insights appreciated

https://github.com/pandas-dev/pandas/blob/12ac43f98c6d3ce9c0c48ed75b80346881c1e4d4/pandas/_libs/algos_rank_helper.pxi.in#L167

**Comment by jreback** [comment]
yeah I think this could be disallowed.

**Comment by WillAyd** [comment]
Fixed as part of #19481
